backend/module/chaos/redisexperimentation/xds/redis_runtime_generation.go

```diff
@@ -0,0 +1,106 @@
+package xds
+
+import (
+	"fmt"
+
+	"go.uber.org/zap"
+
+	experimentation "github.com/lyft/clutch/backend/api/chaos/experimentation/v1"
+	redisexperimentation "github.com/lyft/clutch/backend/api/chaos/redisexperimentation/v1"
+	"github.com/lyft/clutch/backend/service/chaos/experimentation/experimentstore"
+)
+
+const (
+	// REDIS FAULT
+	// a given downstream service to the upstream redis cluster faults
+	RedisLatencyPercentageWithDownstream = `%s.%s.delay.fixed_delay_percent`
+	RedisErrorPercentageWithDownstream   = `%s.%s.delay.error_percent`
+)
+
+func GetEnforcingCluster(experiment *experimentation.Experiment, logger *zap.SugaredLogger) (string, error) {
+	redisFaultConfig := &redisexperimentation.FaultConfig{}
+	if !maybeUnmarshalRedisFaultTest(experiment, redisFaultConfig) {
+		logger.Errorw("Invalid redis fault config", "config", redisFaultConfig)
+		return "", nil
+	}
+
+	upstreamCluster, _ := getRedisClusterPair(redisFaultConfig)
+	return upstreamCluster, nil
```

Comment:
Yes. It should be enforced by downstream as described by `RedisLatencyPercentageWithDownstream`

Reply:
Oh I see what you mean. I'm returning the wrong cluster here 😓 .

Reply:
So actually `upstreamCluster` above is actually the downstreamCluster because `getRedisClusterPair` returns (downstream,upstream). I'll update the variable name to remove the confusion.